R/DataMart.R

```diff
@@ -148,9 +148,12 @@ close.qpostcon = function(con, ...)
                 encode = "raw",
                 body = upload_file(tmpfile)))
 
-    if (inherits(res, "try-error") || res$status_code != 200)
+    if (res$status_code == 413) {
+        stopBadRequest(res, "Could not write to Displayr Cloud Drive. Data to write is too large.")
+    }
+    else if (inherits(res, "try-error") || res$status_code != 200)
```

Comment:
I think you should swap the order here. If there is an error in the `try` statement above. It needs to be caught first before you attempt inspect the `status_code` element. The output of `res` will be an atomic vector which you cannot inspect in that way and will likely throw another error instead saying `res$status_code : $ operator is invalid for atomic vectors`.

Reply:
You could probably add a unit test for this using `with_mock` to make sure it behaves as expected.

Reply:
Well caught! I've changed it to check properly and removed unnecessary braces. 

I'll pass on those unit tests for now, since it's testing at a pretty granular level and shouldn't have much impact anyway.